classwork/сlasswork2.1/main.cpp

```diff
@@ -0,0 +1,89 @@
+#include <iostream>
+#include <unordered_map>
+#include <fstream>
+#include <string>
+#include <print>
+
+class UserStatistics {
+public:
+    UserStatistics() {
+        loadFromFile();
+    }
+
+    ~UserStatistics() {
+        saveToFile();
+    }
+
+    std::string greetUser(const std::string& name) {
+        if (name == "bread") {
+            userStats.clear();
+            saveToFile();
+            return "All history has been exterminated.";
+        }
+
+        if (userStats.find(name) == userStats.end()) {
+            userStats[name] = 1;
+            return "Welcome, " + name + "!";
+        } else {
+            int count = ++userStats[name];
+            return "Hello again(" + std::to_string(count) + "), " + name + "!";
+        }
+    }
+
+    std::string resetUser(const std::string& name) {
+        auto it = userStats.find(name);
+        if (it != userStats.end()) {
+            userStats.erase(it);
+            return "Statistics for " + name + " have been reset.";
+        } else {
+            return "No statistics to reset for " + name + ".";
+        }
+    }
+private:
+    std::unordered_map<std::string, int> userStats;
+    const std::string fileName = "user_stats.txt";
+
+    void loadFromFile() {
+        std::ifstream file(fileName);
+        if (file.is_open()) {
+            std::string name;
+            int count;
+            while (file >> name >> count) {
+                userStats[name] = count;
+            }
+            file.close();
+        }
+    }
+
+    void saveToFile() {
+        std::ofstream file(fileName);
+        if (file.is_open()) {
+            for (const auto& entry : userStats) {
+                file << entry.first << " " << entry.second << "\n";
```

Comment:
std::println can do the magic
`std::println(file, "{} {}", entry.first, entry.second);`
Actually std::print(ln) can print to any ostream. By default it uses stdout so `std::println("Test")` is the same as `std::println(stdout, "Test")` but you can use stderr stream as an. argument or any ostream based object like file, string stream and so on.